iana/hwtypes.go

```diff
@@ -1,7 +1,7 @@
 package iana
 
 // HWType is a hardware type as per RFC 2132 and defined by the IANA.
-type HWType uint8
+type HWType uint16
```

Comment:
? I don't think this is right? And it might break ipv4?

Reply:
This is definitely right for dhcpv6 (and ARP, for that matter, which also has 16bits for this field), see this table (referenced in RFC8415 in https://tools.ietf.org/html/rfc8415#section-11.2): https://www.iana.org/assignments/arp-parameters/arp-parameters.xml#arp-parameters-2, there are assigned hardware types for values 256 and 257

However yes it seems it would break bootp/v4, we need to add an explicit conversion to uint8 there I guess